File: backend/src/main/java/com/fsb/contactmanagement/dao/entities/Offre.java
package com.fsb.contactmanagement.dao.entities;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Offre {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long Id;
    @Column(nullable = false)
    private String nom;
    private String description;
    @Column(nullable = false)
    private Double prix;
    private Double remise;
    @Column(nullable = false)
    private String categorie;
    @Column(nullable = false)
    private Integer duration;
}

File: frontend/src/app/shared/offre.ts
export class Offre {
    id?: number;
    nom: string;
    description?: string;
    prix: number;
    remise?: number;
    categorie: string;
    duration: number;
    image: string;

    constructor(
        nom: string,
        prix: number,
        categorie: string,
        duration: number,
        image: string,
        id?: number,
        description?: string,
        remise?: number
    ) {
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.prix = prix;
        this.remise = remise;
        this.categorie = categorie;
        this.duration = duration;
        this.image = image;
    }
}
